### ISSUE TITLE
.yarnrc.yml file doesn't exist in tagged builds

### ISSUE THREAD

**Issue by SaibotMagd** [elabftw/elabimg#37]
# Describe the bug
When building the docker image from latest tagged build (today it is 4.9.0 see here: https://github.com/elabftw/elabftw/releases/latest) with DOCKER_BUILDKIT=1 docker build --build-arg ELABFTW_VERSION=4.9.0 -t elabftw/elabimg:4.9.0 . 
As descriped in the readme file. The build will fail because there's no .yarnrc.yml file in the https://github.com/elabftw/elabftw/tree/4.9.0 repository.

Check: https://github.com/elabftw/elabftw/tree/4.9.0 there's no .yarnrc.yml file so it have to fail.

- Version of eLabFTW (visible in Sysadmin panel): 4.9.0 also tried 4.8.6
- Server operating system (e.g: Ubuntu, CentOS): Ubuntu LTS 20
- Client operating system (e.g: MacOS, Windows, GNU+Linux): Ubuntu LTS 20
- Browser (e.g. firefox, chrome): latest Firefox

Don't build from tagged release, build from master or latest: https://github.com/elabftw/elabftw/tree/master
So the command: DOCKER_BUILDKIT=1 docker build --build-arg ELABFTW_VERSION=master -t elabftw/elabimg:master .

Since i don't understand any of these things, this could also be the usual behavior and not a bug. In that case, please understand, but I was confused by the instructions and it took me an hour to find the problem.

**Comment by NicolasCARPi** [comment]
Hello,

I'm sorry for the confusion. A particular `elabftw` version **must** be built with the corresponding `elabimg` version.

Newer `elabimg` versions expect Yarn v4 with a `.yarnrc`, but that's only since `elabftw` version 5.0. So for `4.9.0` you'll need to build with the corresponding `elabimg` version. (https://github.com/elabftw/elabimg/releases/tag/4.9.0).

To make things easier, the `elabimg` versions are tagged with the same tag as the elabftw version.

So to build eLabFTW X.Y.Z you need to built from `elabimg` X.Y.Z!

This works:

~~~
git clone -b 4.9.0 https://github.com/elabftw/elabimg
cd elabimg
docker buildx build --build-arg ELABFTW_VERSION=4.9.0 -t elabftw/elabimg:4.9.0 .
~~~


> Don't build from tagged release, build from master or latest:

On the opposite, build from tagged releases!

Finally, end users should not have to build the image, could you share why you want to build it yourself?

**Comment by SaibotMagd** [comment]
> I'm sorry for the confusion. A particular `elabftw` version **must** be built with the corresponding `elabimg` version.
Sure, that's what I thought and what I did, did'n I?

> Newer `elabimg` versions expect Yarn v4 with a `.yarnrc`, but that's only since `elabftw` version 5.0. So for `4.9.0` you'll need to build with the corresponding `elabimg` version. (https://github.com/elabftw/elabimg/releases/tag/4.9.0).
I don't know what that means. If the file is not available, the build aborts and I couldn't recognized a Version 5 in the tagged releases.

> This works:
> 
> ```
> git clone -b 4.9.0 https://github.com/elabftw/elabimg
> cd elabimg
> docker buildx build --build-arg ELABFTW_VERSION=4.9.0 -t elabftw/elabimg:4.9.0 .
> ```
buildx wasn't mention in the documentation and I don't know about it, as I'm not really familar with docker.

> Finally, end users should not have to build the image, could you share why you want to build it yourself?
Aim I called an "end user"? I'm suposed to develop a sample tracking extension for elabFTW (https://github.com/SaibotMagd/dmtxSampleCreator) and for this I need an elabFTW testserver. Since we are a research institute and do not use elabftw, there is no one who would set up a testserver. That's why I followed the instructions and did it myself. 

Btw. the server is running and I got the API connection to integrate a connection to elabFTW into my sample tracker.

**Comment by NicolasCARPi** [comment]
> what I did, did'n I?

No, otherwise it would have worked. You probably ran the correct command but on the `master` branch of `elabimg`, instead of the tagged commit.

> I don't know what that means. If the file is not available, the build aborts and I couldn't recognized a Version 5 in the tagged releases.

It means elabimg version 5 cannot build elabftw version 4.

> buildx wasn't mention in the documentation 

The [documentation](https://github.com/elabftw/elabimg#building-this-image) indicates: `DOCKER_BUILDKIT=1 docker build... ` which is the same as `docker buildx build ...` (I realize it is not evident for someone unfamiliar with docker)

> Aim I called an "end user"?

Yes, you are a consumer of the docker image, you never need to build it yourself as you can simply do `docker pull elabftw/elabimg:4.9.0` (don't even have to do that because docker-compose will do it for you).

> I'm suposed to develop a sample tracking extension for elabFTW 

Then you're not a end user but a developer, and must use this documentation: https://doc.elabftw.net/contributing.html#contributing-to-the-code. Unless you don't plan to modify eLabFTW and simply want a running instance to play with, in that case you're good!
